lib/nexpose/scan_template.rb

```diff
@@ -191,82 +191,82 @@ def enable_ip_stack_fingerprinting=(enable)
       end
     end
 
-    # Enable/diisable ICMP device discovery
-    # @param [Boolean] enable or disable ICMP device discovery
-    def enable_icmp_discovery=(enable)
+    # Enable/disable ICMP device discovery
+    # @param [Boolean] enable Enable or disable ICMP device discovery
+    def enable_icmp_device_discovery=(enable)
       icmp = REXML::XPath.first(@xml, 'ScanTemplate/DeviceDiscovery/CheckHosts/icmpHostCheck')
       icmp.attributes['enabled'] = (enable ? 1 : 0)
     end
 
-    # Add custom TCP ports to scan for device discovery
-    # @param [Array] ports to scan for device discovery
-    def tcp_discovery_ports=(ports)
+    # Enable/disable TCP device discovery
+    # @param [Boolean] enable Enable or disable TCP device discovery
+    def enable_tcp_device_discovery=(enable)
       tcp = REXML::XPath.first(@xml, 'ScanTemplate/DeviceDiscovery/CheckHosts/TCPHostCheck')
-      REXML::XPath.first(tcp, './portList').text = ports.join(',')
+      tcp.attributes['enabled'] = (enable ? 1 : 0)
     end
 
-    # Enable/disable TCP device discovery
-    # @param [Boolean] enable or disable TCP device discovery
-    def enable_tcp_discovery=(enable)
+    # Set custom TCP ports to scan for device discovery
+    # @param [Array] ports Ports to scan for device discovery
```

Comment:
Actually the Rubydoc website doesn't even show `@param` stuff correctly, so not sure what to think of that...